src/storage/src/render/persist_sink.rs

```diff
@@ -52,7 +51,7 @@ pub fn render<G>(
 
     let mut input = persist_op.new_input(&source_data.inner, Exchange::new(move |_| hashed_id));
 
-    let shared_frontier = Rc::clone(&storage_state.source_uppers[&src_id]);
+    let current_upper = Rc::clone(&storage_state.source_uppers[&src_id]);
```

Comment:
does this need to be instantiated here anymore?

Reply:
Hm, can you elaborate a bit? Why wouldn't it be needed?